### PR TITLE
Ssh personal environment

### PR DESCRIPTION
**Pull request by schlomo**
- refactor Docker build environment to have all configs that go into the container in a commen top-level dir
- Export all service logs to Docker stdout (could not see bottle errors)
- Clean up Dockerfile
- Add ssh client
- Add screen with configuration
- Add tmux
- Add python-virtualenv to complete the pip installation into something usable
- Add zsh

### CODE REVIEW

File: configs/supervisord.conf [hunk 1]
```diff
@@ -10,14 +10,14 @@ stdout_logfile_maxbytes=0
...
-stdout_logfile=/var/log/supervisor/cron.log
+stdout_logfile=/dev/stderr
 stdout_logfile_maxbytes=0
 
 [program:c-bastion]
 command=python /usr/local/bin/c-bastiond
-stderr_logfile=/var/log/supervisor/c-bastion_error.log
+stderr_logfile=/dev/stderr
 stderr_logfile_maxbytes=0
-stdout_logfile=/var/log/supervisor/c-bastion.log
```
Comment (esc) [inline configs/supervisord.conf:22]: Where do the logs end up when the whole thing is deployed?

**Comment by esc** [comment]
Travis had a hickup, I suppose, restarting.

File: configs/supervisord.conf (hunk 1 above)
Comment (schlomo) [inline configs/supervisord.conf:22]: In the stdout of the Docker container as per our separation of concern design. Log shipping is the job of the Docker host, not the container.

**Comment by esc** [comment]
it seems like the docker integration is broken.

**Comment by schlomo** [comment]
It hangs on `docker push`. IMHO doing a `docker push` on a PR build is wrong, that should only happen for builds from master.

**Comment by esc** [comment]
Yeah, see #6

**Comment by esc** [comment]
Last time I tried, the `travis.yaml` wasn't flexible enough to allow this.

**Comment by esc** [comment]
also, this has been working all along, so it is surprising that the upload is breaking now.

**Comment by esc** [comment]
Regarding the logging stuff, I'd like to reject this change for the time being, at least until the logging situation in Taupage has been cleaned up. This is just a temporary rejection. How do you feel about that?

**Comment by esc** [comment]
Since #6 has been fixed, please rebase against master and I'll merge it.

**Comment by esc** [comment]
Wahay, travis is green!

**Comment by esc** [comment]
Let me know, if you figure out where the logs go in Taupage.

**Comment by esc** [comment]
Also, you'll need to wait a few minutes before Travis builds the master and the image becomes available at DockerHub.

**Comment by schlomo** [comment]
`/var/log/application.log` seems to be the default place for the logs on
Taupage.

**Pull request merged**